server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -75,11 +75,15 @@
 
   private final AccumuloServerContext context;
   private final VolumeManager fs;
-  private final boolean useTrash;
+  private final RemovalPolicy removalPolicy;
   private final LiveTServerSet liveServers;
   private final WalStateManager walMarker;
   private final Iterable<TabletLocationState> store;
 
+  enum RemovalPolicy {
+    ARCHIVE, ARCHIVE_TRASH, TRASH, DELETE
```

Comment:
I don't think ARCHIVE and ARCHIVE_TRASH are needed, since TRASH basically is a temporary archive already.